src/components/views/dialogs/spotlight/SpotlightDialog.tsx

```diff
@@ -480,12 +481,15 @@ const SpotlightDialog: React.FC<IProps> = ({ initialText = "", initialFilter = n
         // eslint-disable-next-line
     }, [results, filter]);
 
-    const viewRoom = (roomId: string, persist = false, viaKeyboard = false) => {
+    const viewRoom = (room: {room_id: string}|{room_alias: string}, persist = false, viaKeyboard = false) => {
```

Comment:
This could just take a `roomIdOrAlias: string` as some other methods do and it could differentiate them based on the leading sigil

Reply:
Yeah, I went this way because the viewRoom API seems to want it explicitly, but happy to go the other way.

Reply:
Yes, the ViewRoom dispatch will need it in the right field but it could be this method that turns `roomIdOrAlias` into the right field

Reply:
Ah of course if we did want to support the persist behaviour we'd want to include both values.

Reply:
You can provide both over the dispatch if you have both and it'll then be able to skip an explicit alias->room id lookup for even bettererer efficiency

Reply:
I've since kept the object as it felt like having persist would be a good idea. We could chuck the roomAlias in as a 4th parameter to the function instead of an object though?

Reply:
Totally up to your preference